### PR TITLE
Turn on racetest for all presubmits

### PR DESCRIPTION
**Pull request by xiaolanz**
addresses #4235

We will watch submission queue for a few days before turn on "required"

### CODE REVIEW

**Comment by ldemailly** [comment]
looks like there are still failures?

**Comment by xiaolanz** [comment]
Looks like some tests are being flaking.

/retest

**Comment by ldemailly** [comment]
circle doesn't understand "/retest" you have to click the button in the UI

**Comment by ldemailly** [comment]
https://github.com/istio/istio/wiki/Working-with-CircleCI

**Comment by ldemailly** [comment]
still not passing :(

**Comment by xiaolanz** [comment]
@wattli  all other racetest passes beside i'm getting the following error in ci/circleci: racetest 

do you have some idea on it? thank

FAIL	istio.io/istio/security/tests/integration/secretCreationTest	0.565s
make: *** [security-racetest] Error 1
Makefile:524: recipe for target 'security-racetest' failed
Exited with code 2

**Comment by xiaolanz** [comment]
PTAL

File: .circleci/config.yml [hunk 1]
```diff
@@ -692,3 +691,6 @@ workflows:
       - e2e-simple:
           requires:
             - build
+      - racetest:
```
Comment (costinm) [inline .circleci/config.yml]: No longer need dependencies - but you may run it after test.

**Comment by xiaolanz** [comment]
@costinm this turns on for presubmit but doesn't block. So i can find the flakies.

**Comment by sebastienvas** [comment]
We should NEVER move a whole suite to flaky when only one test is flaky. Instead we should create an ignore file, where you put the flaky test, and file a bug when this happen. Preventing all the tests from running is just an open doors for more coming bugs.

**Comment by sebastienvas** [comment]
Note that you can skip a test for flaky directly in the test which is even better than the ignore file.

**Comment by costinm** [comment]
Yes - it's fine to ignore/skip tests from the suite if a bug is filled. But so far this has not been done for 
racetests. 

Where the test is running should have no impact on fixing the bugs - all tests should upload to the 
dashboard, and failing tests need to be investigated and fixed. 

The cirlceCI presubmit is intended to contain a subset of the tests suites, with <15 min run time and very reliable. There are 2 periodic runners and a nightly build, running other suites - and prow. There are cases where the presubmit will not catch bugs that are covered by longer-running suites - passsing
presubmit doesn't mean a change is perfect, or that if a bug is found in the load test or periodic suites 
it is lower priority.

**Comment by sebastienvas** [comment]
Let's discuss this elsewhere. The 15 mn requirement only makes sense when you know what are testing for. Your presubmit time should be defined by the % of postsubmit failure not just by a random number.  15mn is something we want to reach eventually, but not a hard stop.

**Comment by ldemailly** [comment]
back to the topic of the PR, racetest is still failing:

for instance https://circleci.com/gh/istio/istio/52665

**Comment by xiaolanz** [comment]
This is not enforced so new tests can break racetest. 

TestKeyCertBundleRotator is a regression. Filed bug #4385

**Comment by ldemailly** [comment]
if it's not enforced it'll get back or stay red forever. though conversely it has to be non flaky green to be enforced. I know you are fighting a tide there joy, but you're almost there... keep pushing!

**Comment by xiaolanz** [comment]
Yeah, it's a deliberate art to find and exclude all flaky tests while identifying and fixing the real regressions in a timely fashion.

**Pull request merged**